com.unity.ml-agents/Tests/Editor/MLAgentsEditModeTest.cs

```diff
@@ -524,12 +525,11 @@ public void TestMaxStepsReset()
 
             for (var i = 0; i < 15; i++)
             {
-                // We expect resets to occur when there are maxSteps actions since the last reset (and on the first step)
-                var expectReset = agent1.agentActionCallsSinceLastReset == maxStep || (i == 0);
+                Debug.Log(i);
+                // We expect resets to occur when there are maxSteps actions since the last reset 
+                var expectReset = agent1.agentActionCallsSinceLastReset == maxStep;
```

Comment:
Hang on, this test looks broken. `var previousNumResets` used to be the number of resets before stepping. But now we assert on its value before stepping the Academy. So `previousNumResets + 1, agent1.agentResetCalls` can never be true, which means expectReset must never be true. So you're basically testing `agent1.agentResetCalls== agent1.agentResetCalls1`

Reply:
Re-wrote the test that was broken

Reply:
Looks pretty good. I made a few additions on https://github.com/Unity-Technologies/ml-agents/pull/3467 - can you merge that if it seems reasonable?